src/naming/naming.interface.ts

```diff
@@ -0,0 +1,4 @@
+export interface NameService {
+  getSNSNameByPublicKey(publicKey: string): Promise<String>;
+  getTwitterHandler(publicKey: string): Promise<String>;
+}
```

Comment:
this looks like an uber interface which doesn't hide implementation details. I'd suggest to use some kind of polymorphism and hide implementation details. 
E.g. have generic `NameResolver` and different implementations like `SNSNameResolver` and `CardinalNameResolver`. In this way it'll be much easier to test code and have different implementations at runtime

Reply:
Yes, you're right. Also thought about it